codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/Encoder.java

```diff
@@ -133,22 +135,23 @@ public void encodeHeader(ByteBuf out, CharSequence name, CharSequence value, boo
     /**
      * Set the maximum table size.
      */
-    public void setMaxHeaderTableSize(ByteBuf out, int maxHeaderTableSize) {
-        if (maxHeaderTableSize < 0) {
-            throw new IllegalArgumentException("Illegal Capacity: " + maxHeaderTableSize);
+    public void setMaxHeaderTableSize(ByteBuf out, long maxHeaderTableSize) {
+        if (maxHeaderTableSize < MIN_HEADER_TABLE_SIZE || maxHeaderTableSize > MAX_HEADER_TABLE_SIZE) {
+            throw new IllegalArgumentException("maxHeaderTableSize is invalid: " + maxHeaderTableSize);
         }
         if (capacity == maxHeaderTableSize) {
             return;
         }
         capacity = maxHeaderTableSize;
         ensureCapacity(0);
-        encodeInteger(out, 0x20, 5, maxHeaderTableSize);
+        // Casting to integer is safe as we verified the maxHeaderTableSize is a valid unsigned int.
+        encodeInteger(out, 0x20, 5, (int) maxHeaderTableSize);
```

Comment:
@normanmaurer - it seems like this may be problematic if `maxHeaderTableSize > Integer.MAX_INT`. Do we have a unit test covering this case?